### PR TITLE
Add STM32F103xD, STM32F103xG series and fix STM32F103xE startup file

### PR DESCRIPTION
**Pull request by jancotipsarevic**
### Documentation 

----------------------------------------------------------------------------------------------------------------
### Pull request type 

    [x] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results 

    [x] No Tests required for this change (E.g docs only update)
    [] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers 

----------------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: targets/TARGET_STM/TARGET_STM32F1/TARGET_STM32F103xD/TOOLCHAIN_IAR/stm32f103xd.icf [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/* Linker script to configure memory regions.
```
Comment (multiplemonomials) [inline targets/TARGET_STM/TARGET_STM32F1/TARGET_STM32F103xD/TOOLCHAIN_IAR/stm32f103xd.icf:1]: Mbed OS does not support IAR anymore, so the IAR linker scripts and startup files don't need to be added

**Comment by multiplemonomials** [comment]
Hi!  Thank you for the contribution.

Any way you could provide some more detail on the motivation for this change?  What boards are these chips used in?  Or are they for custom boards only?

**Comment by JohnK1987** [comment]
I believe it is in context of this question https://github.com/mbed-ce/mbed-os/discussions/124

These MCUs [were added to ARM MbedOS in January](https://github.com/ARMmbed/mbed-os/tree/master/targets/TARGET_STM/TARGET_STM32F1), exactly 2 days after last [Upstream merge](https://github.com/mbed-ce/mbed-os/pull/117), but many more changes were done and Mbed OS6.17 was also released.

File: targets/TARGET_STM/TARGET_STM32F1/TARGET_STM32F103xD/TOOLCHAIN_IAR/stm32f103xd.icf (hunk 1 above)
Comment (jancotipsarevic) [reply targets/TARGET_STM/TARGET_STM32F1/TARGET_STM32F103xD/TOOLCHAIN_IAR/stm32f103xd.icf:1]: OK, I was not aware about that, Should I do something or you could manage it?

**Comment by jancotipsarevic** [comment]
> Hi! Thank you for the contribution.
> 
> Any way you could provide some more detail on the motivation for this change? What boards are these chips used in? Or are they for custom boards only?

I need it for custom boards. My first intention is to port one existing application (STM32F103xD) using Mbed Studio to Mbed-CE+CLion, and I plan to start a new application using STM32F103VG.

**Comment by jancotipsarevic** [comment]
> I believe it is in context of this question #124
> 
> These MCUs [were added to ARM MbedOS in January](https://github.com/ARMmbed/mbed-os/tree/master/targets/TARGET_STM/TARGET_STM32F1), exactly 2 days after last [Upstream merge](https://github.com/mbed-ce/mbed-os/pull/117), but many more changes were done and Mbed OS6.17 was also released.

It would be great to update to 6.17 , it of course will solve my issue, but I could imagine it is a huge work, I will be more than happy if this PR is accepted to start working seriously with Mbed-CE asap.

**Comment by multiplemonomials** [comment]
Oh so this code is copied straight from Mbed?

**Comment by jancotipsarevic** [comment]
> Oh so this code is copied straight from Mbed?

Yes, it is.

**Comment by multiplemonomials** [comment]
In that case, this can be greenlighted!  I just thought that you wrote all this yourself so I wanted some more background info.

**Review by multiplemonomials — APPROVED** [review]

**Comment by jancotipsarevic** [comment]
> In that case, this can be greenlighted! I just thought that you wrote all this yourself so I wanted some more background info.

Sorry, next time I´ll be more specific. Thank you very much.

**Pull request merged**